src/compiler/enums.js

```diff
@@ -0,0 +1,195 @@
+ */
+const ValueType = {
+    NUMBER: 1,
+    STRING: 2,
+    BOOLEAN: 3,
+    UNKNOWN: 4,
+    NUMBER_OR_NAN: 5,
+}
+
+/**
+ * Enum for the opcodes of the stackable blocks used in the IR AST.
+ * @readonly
+ * @enum {string}
+ */
+const BlockOpcode = {
+    NOP: "noop",
+
+    CONTROL_IF_ELSE: "control.if",
+    CONTROL_CLONE_CREATE: "control.createClone",
+    CONTROL_CLONE_DELETE: "control.deleteClone",
+    CONTROL_WHILE: "control.while",
+    CONTROL_FOR: "control.for",
+    CONTROL_REPEAT: "control.repeat",
+    CONTROL_STOP_ALL: "control.stopAll",
+    CONTROL_STOP_OTHERS: "control.stopOthers",
+    CONTROL_STOP_SCRIPT: "control.stopScript",
+    CONTROL_WAIT: "control.wait",
+    CONTROL_WAIT_UNTIL: "control.waitUntil",
+    CONTROL_WHILE: "control.while",
+
+    LIST_ADD: "list.add",
+    LIST_INSERT: "list.instert",
+    LIST_REPLACE: "list.replace",
+    LIST_DELETE_ALL: "list.deleteAll",
+    LIST_DELETE: "list.delete",
+    LIST_SHOW: "list.show",
+    LIST_HIDE: "list.hide",
+
+    VAR_SET: "var.set",
+    VAR_SHOW: "var.show",
+    VAR_HIDE: "var.hide",
+
+    EVENT_BROADCAST: "event.broadcast",
+    EVENT_BROADCAST_AND_WAIT: "event.broadcastAndWait",
+
+    LOOKS_EFFECT_SET: "looks.setEffect",
+    LOOKS_EFFECT_CHANGE: "looks.changeEffect",
+    LOOKS_EFFECT_CLEAR: "looks.clearEffects",
+    LOOKS_SIZE_CHANGE: "looks.changeSize",
+    LOOKS_SIZE_SET: "looks.setSize",
+    LOOKS_LAYER_FORWARD: "looks.forwardLayers",
+    LOOKS_LAYER_BACKWARD: "looks.backwardLayers",
+    LOOKS_LAYER_FRONT: "looks.goToFront",
+    LOOKS_LAYER_BACK: "looks.goToBack",
+    LOOKS_HIDE: "looks.hide",
+    LOOKS_SHOW: "looks.show",
+    LOOKS_BACKDROP_NEXT: "looks.nextBackdrop",
+    LOOKS_BACKDROP_SET: "looks.switchBackdrop",
+    LOOKS_COSTUME_NEXT: "looks.nextCostume",
+    LOOKS_COSTUME_SET: "looks.switchCostume",
+
+    MOTION_X_SET: "motion.setX",
+    MOTION_X_CHANGE: "motion.changeX",
+    MOTION_Y_SET: "motion.setY",
+    MOTION_Y_CHANGE: "motion.changeY",
+    MOTION_XY_SET: "motion.setXY",
+    MOTION_IF_ON_EDGE_BOUNCE: "motion.ifOnEdgeBounce",
+    MOTION_STEP: "motion.step",
+    MOTION_ROTATION_STYLE_SET: "motion.setRotationStyle",
+    MOTION_DIRECTION_SET: "motion.setDirection",
+
+    PEN_UP: "pen.up",
+    PEN_DOWN: "pen.down",
+    PEN_CLEAR: "pen.clear",
+    PEN_COLOR_PARAM_SET: "pen.setParam",
+    PEN_COLOR_PARAM_CHANGE: "pen.changeParam",
+    PEN_COLOR_HUE_CHANGE_LEGASY: "pen.legacyChangeHue",
+    PEN_COLOR_HUE_SET_LEGASY: "pen_setPenHueToNumber",
+    PEN_COLOR_SHADE_CHANGE_LEGASY: "pen.legacyChangeShade",
+    PEN_COLOR_SHADE_SET_LEGASY: "pen.legacySetShade",
+    PEN_COLOR_SET: "pen.setColor",
+    PEN_SIZE_SET: "pen.setSize",
+    PEN_SIZE_CHANGE: "pen.changeSize",
+    PEN_STAMP: "pen.stamp",
+
+    SENSING_TIMER_RESET: "timer.reset",
+
+    PROCEDURE_CALL: "procedures.call",
+
+    TW_ADDON_CALL: "addons.call",
+    TW_DEBUGGER: "tw.debugger",
+    TW_VISUAL_REPORT: "visualReport",
+    TW_COMPATIBILITY_LAYER: "compat"
```

Comment:
let's call it ADDON_CALL, DEBUGGER, VISUAL_REPORT, COMPATIBILITY_LAYER (remove the TW_, that's for "TurboWarp Blocks") and move it after NOP (they are special)